node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/NodesResponse.java

```diff
@@ -246,4 +233,29 @@ private String lastElement(String path) {
         return path.substring(lastSlash+1);
     }
 
+    static Optional<String> nodeArchiveUrl(FlagSource flagSource, Node node) {
```

Comment:
This is just temporary, eventually the config servers will create bucket and assign applications to buckets, so this will be stored either on the node itself or in `com.yahoo.vespa.hosted.provision.applications.Application`.

Reply:
Consider adding a TODO for this.